src/react/components/pages/editorPage/editorPage.tsx

```diff
@@ -325,6 +327,30 @@ export default class EditorPage extends React.Component<IEditorPageProps, IEdito
         });
     }
 
+    private async updateProjectTagsFromAsset(asset: IAssetMetadata) {
+        const assetTags = new Set();
+        asset.regions.forEach((region) => region.tags.forEach((tag) => assetTags.add(tag.name)));
+
+        const newTags: ITag[] = this.props.project.tags ? this.props.project.tags.slice() : [];
```

Comment:
can be rewritten as `const newTags = [...this.props.project.tags];`

Reply:
You still have to check if null before otherwise you got a spread error (basically null is non iterable).  So in this case slice or spread are really the same but I'll go with spread as it is well spread in the codebase ;-)